homework-solutions/js/ts/hw-15/task-1.ts

```diff
@@ -0,0 +1,57 @@
+// 1. Создайте интерфейс IVehicle:
+//   Методы:
+//     - getDetails(): string — возвращает информацию о транспортном средстве.
+//     - start(): string — возвращает строку "The vehicle is starting".
+
+interface IVehicle {
+  getDetails(): string;
+  start(): string;
+}
+
+// 2. Создайте абстрактный класс Vehicle, который имплементирует IVehicle:
+//   Реализуйте конструктор с полями:
+//     - make (строка)
+//     - model (строка)
+//   Добавьте методы:
+//     - start, возвращающего строку: "The vehicle {make} {model} is starting.".
+//     - Абстрактный метод getDetails, который нужно реализовать в классах-наследниках.
+
+abstract class Vehicle implements IVehicle {
+  constructor(
+    public make: string,
+    public model: string,
+  ) {}
+
+  start(): string {
+    return `The vehicle ${this.make} ${this.model} is starting.`;
+  }
+
+  abstract getDetails(): string;
+}
+
+// 3. Создайте класс Car, который наследует Vehicle:
+//     - Добавляет поле year (число).
+//     - Реализует метод getDetails, возвращающего строку: "{make} {model}, {year}".
+
+class Car extends Vehicle {
+  constructor(
+    public make: string,
+    public model: string,
+    public year: number,
```

Comment:
Нам не нужно указывать модификаторы свойств в наследниках, если мы передаем их в `super`
Получается при использования модификатора мы создаем свойство в самом классе 'Car`, если свойство новое в случае `year`, тут как раз модификатор пригодится
```suggestion
    make: string,
    model: string,
    public year: number,
```